src/types/plugin-inputs.ts

```diff
@@ -36,6 +36,17 @@ export const userActivityWatcherSettingsSchema = T.Object({
    * Delay to send reminders. 0 means disabled. Any other value is counted in days, e.g. 1,5 days
    */
   warning: thresholdType({ default: "3.5 days" }),
+  /**
+   * Define how different organizations, users or specific repositories
+   * should be watched. Use the following format:
+   * 
+   * - "ubiquibot" - all repositories in the organization
+   * - "ubiquity/ubiquibot-logger" - specific repository
+   */
+  watch: T.Object({
+    optIn: T.Array(T.String()),
+    optOut: T.Array(T.String()),
+  }),
```

Comment:
I'm not sure if you meant to implement this new watch field entirely separate from plugins or in settings?
```
watch:
  optIn: ...

plugins:
  - uses:
    - plugin: ubiquibot/user-activity-watcher@main
      with:
        unassignUserThreshold: "5 minutes"
        sendRemindersThreshold: "1 minutes"
```
OR
```
plugins:
  - uses:
    - plugin: ubiquibot/user-activity-watcher@main
      with:
        unassignUserThreshold: "5 minutes"
        sendRemindersThreshold: "1 minutes"
        watch:
          optIn: ...
```
because the code you wrote will only work for the latter.

Also in the spec there's an example where `optIn` includes other organizations - how does it make sense for one org to turn on this plugin for another org - they might be related orgs but we don't know that?

Reply:
Option two where it's built into the plugin settings afaik the global settings are pretty much deprecated? So it'll be defined at the org level and less likely to be customized per repo (this value at least perhaps the thresholds might be)

> - how does it make sense for one org to turn on this plugin for another org - they might be related orgs but we don't know that?

How can we validate that there is ownership between the two? 

- could we use the `installID` or would it be different for each org regardless of whether I give my bot access to all 4 of my orgs?
- even if the above is true, I see that they might still be different
- same admins/owner can't be guaranteed

We could restrict updates to only the payload organization and from this config we can tell which ones

Reply:
Installation ID corresponds to each unique organization it is added to.

Reply:
> So the opt system is intended to be used in the organization config only? Or alternatively if defined within repo config, then does it mean that only the repo it is defined in can invoke the plugin?

My understanding is we don't use global config settings anymore instead it's through each plugin's settings. I don't see why this plugin would be customized at the repo level realistically but certainly not `watch` as what would be the need?

Is the problem that if the other items are modified at the repo level then org level `watch` would be overridden?

> > same admins/owner can't be guaranteed

> Why not?

> Because we should treat every org separately

In our case yeah it can but it doesn't seem like it'd be a good blanket approach.

> So at risk of a possible security leak, we can define the installation IDs in the private config file potentially to prove ownership?

At the cost of simplicity, couldn't we encrypt it and then use their actual installID to decrypt it? This way it's stored and transmitted securely then the kernel/this plugin could verify?

Reply:
Encryption aside, if the `.ubiquibot-config` repository is private then I'm assuming the data won't be leaked, right?

Reply:
I believe so yeah but haven't you mentioned before that repo should be public?

Reply:
Public maybe for forking to set up a new organization but the real ones in use should be private so that we can store sensitive information if we want I suppose. 

I don't recall why I would request for them to be public.

Reply:
I think linking orgs will just complicate things. It's probably easiest to just keep this plugin at org level and specify which repos to watch or all repos. We can add linking orgs at a later stage if that's needed.